test/tstKokkosToolsAnnotations.cpp

```diff
@@ -0,0 +1,110 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include <ArborX_LinearBVH.hpp>
+
+#include <impl/Kokkos_Profiling.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <string>
+
+#include "Search_UnitTestHelpers.hpp"
+
+BOOST_AUTO_TEST_SUITE(KokkosToolsAnnotations)
+
+namespace tt = boost::test_tools;
+
+template <typename T>
+struct TreeTypeTraits;
+
+template <typename... DeviceTypes>
+struct TreeTypeTraits<std::tuple<DeviceTypes...>>
+{
+  using type = std::tuple<ArborX::BVH<DeviceTypes>...>;
+};
+
+using TreeTypes = typename TreeTypeTraits<ARBORX_DEVICE_TYPES>::type;
+
+bool isPrefixedWith(std::string const &s, std::string const &prefix)
+{
+  return s.find(prefix) == 0;
+}
+
+BOOST_AUTO_TEST_CASE(is_prefixed_with)
+{
+  BOOST_TEST(isPrefixedWith("ArborX::Whatever", "ArborX"));
+  BOOST_TEST(!isPrefixedWith("Nope", "ArborX"));
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(kernels_prefixed, Tree, TreeTypes)
+{
+  auto const callback = [](char const *label, uint32_t, uint64_t *) {
+    std::cout << label << '\n';
+    BOOST_TEST((isPrefixedWith(label, "ArborX_") ||
+                isPrefixedWith(label, "Kokkos::View::") ||
+                isPrefixedWith(label, "Kokkos::ViewFill-1D")));
+  };
+  Kokkos::Tools::Experimental::set_begin_parallel_for_callback(callback);
+  Kokkos::Tools::Experimental::set_begin_parallel_scan_callback(callback);
+  Kokkos::Tools::Experimental::set_begin_parallel_reduce_callback(callback);
+
+  // BVH::BVH
+
+  { // default constructed
+    Tree t;
```

Comment:
Maybe use a more meaningful `tree` or `bvh` instead of `t` throughout?

Reply:
Sure